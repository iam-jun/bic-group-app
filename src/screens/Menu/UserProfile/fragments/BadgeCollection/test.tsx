import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
