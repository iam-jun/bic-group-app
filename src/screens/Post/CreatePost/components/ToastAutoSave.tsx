import {View, Animated, StyleSheet, Easing} from 'react-native';
import React, {useImperativeHandle, useRef} from 'react';
import {ExtendedTheme, useTheme} from '@react-navigation/native';
import Icon from '~/beinComponents/Icon';

import Text from '~/beinComponents/Text';
import {isAndroidAnimated} from '../helper';
import spacing from '~/theme/spacing';

interface Props {
  viewRef: any;
  visible: boolean;
}

const ToastAutoSave = ({viewRef, visible}: Props) => {
  const theme: ExtendedTheme = useTheme();
  const styles = themeStyles(theme);
  const toastHeightAnimated = useRef(new Animated.Value(0)).current;
  const isAnimated = isAndroidAnimated();

  const startAnimation = (height: number, toastHeight: number) => {
    toastHeightAnimated.stopAnimation();
    Animated.timing(toastHeightAnimated, {
      toValue: toastHeight,
      duration: toastHeight === 0 ? 0 : 50,
      useNativeDriver: false,
      easing: Easing.ease,
    }).start();
  };

  useImperativeHandle(viewRef, () => ({
    startAnimation,
  }));

  return (
    <Animated.View
      style={
        isAnimated ? {overflow: 'hidden', height: toastHeightAnimated} : {}
      }>
      {visible && (
        <View style={styles.toastAutoSave}>
          <Icon
            isButton
            iconStyle={styles.iconToastAutoSave}
            style={styles.iconToastAutoSaveContainer}
            size={18}
            icon="Save"
            tintColor={theme.colors.gray50}
          />
          <Text.BodyS useI18n style={styles.textToastAutoSave}>
            post:auto_saved
          </Text.BodyS>
        </View>
      )}
    </Animated.View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const {colors} = theme;

  return StyleSheet.create({
    toastAutoSave: {
      flexDirection: 'row',
      alignItems: 'center',
      backgroundColor: colors.white,
      paddingHorizontal: spacing.padding.large,
      marginBottom: spacing.margin.base,
    },
    iconToastAutoSaveContainer: {marginRight: spacing.margin.tiny},
    iconToastAutoSave: {
      padding: 2,
      borderRadius: 6,
      backgroundColor: colors.neutral80,
    },
    textToastAutoSave: {color: colors.gray50},
  });
};

export default ToastAutoSave;
