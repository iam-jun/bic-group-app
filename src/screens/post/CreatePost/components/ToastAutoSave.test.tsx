import React from 'react';

import { renderWithRedux } from '~/test/testUtils';
import ToastAutoSave from './ToastAutoSave';

describe('ToastAutoSave component', () => {
  it('should render correctly', () => {
    const wrapper = renderWithRedux(<ToastAutoSave visible />);

    expect(wrapper).toMatchSnapshot();
  });
});
