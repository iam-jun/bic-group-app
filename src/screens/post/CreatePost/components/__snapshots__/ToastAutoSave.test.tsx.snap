// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastAutoSave component should render correctly 1`] = `
<View
  collapsable={false}
  style={Object {}}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flexDirection": "row",
        "marginBottom": 12,
        "paddingHorizontal": 16,
      }
    }
  >
    <View
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      hitSlop={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {
              "marginRight": 4,
            },
            Object {
              "backgroundColor": undefined,
            },
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
              Object {
                "borderRadius": 6,
                "padding": 2,
              },
            ]
          }
        >
          <View />
        </View>
      </View>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#212745",
            "fontFamily": "BeVietnamPro-Medium",
            "fontSize": 13,
          },
          Object {
            "color": "#353E6E",
          },
          undefined,
        ]
      }
    >
      Post auto-saved
    </Text>
  </View>
</View>
`;
